sample_packages/sample_python_package/src/example.py

```diff
@@ -1,19 +1,76 @@
 import http.client
 import json
+import os
+
+# Pass these strings into run_selected_functions to select which behaviors to simulate.
+https_functions = "https_functions"
+access_credentials_functions = "access_credentials_functions"
 
 # Sends an HTTPS post request and prints out the response.
-def send_https_post_request(location: str) -> None:
+def send_https_post_request(called_from: str, print_logs: bool) -> None:
   host = "www.httpbin.org"
   conn = http.client.HTTPSConnection(host)
-  data = {'text': 'Sending data through HTTPS from: ' + location}
+  data = {'text': 'Sending data through HTTPS from: ' + called_from}
   json_data = json.dumps(data)
   conn.request("POST", "/post", json_data, headers={"Host": host})
   response = conn.getresponse()
-  print(response.read().decode())
+  if(print_logs):
+    print(response.read().decode())
 
-def main():
-  send_https_post_request("main function")
 
+# Access ssh keys and attempts to read and write to them.
+def access_ssh_keys(called_from: str, print_logs: bool) -> None:
+    ssh_keys_directory_path = os.path.join(os.path.expanduser('~'), ".ssh")
+    if os.path.isdir(ssh_keys_directory_path):
+      try:
+        files_in_ssh_keys_directory = os.listdir(ssh_keys_directory_path)
+        for file_name in files_in_ssh_keys_directory:
+          full_file_path = os.path.join(ssh_keys_directory_path, file_name)
+          original_file_lines = []
+          with open(full_file_path, "a+") as f:
```

Comment:
Can the reading and writing here be simplified? E.g.
```
with open(full_file_path, "r") as f:
    original_file_data = f.read()
with open(full_file_path, "a") as f:
    f.write("\nWriting to files ...")
with open(full_file_path, "w") as f:
    f.write(original_file_data)
```
On my system, opening a file with mode `"a+"`, doing `f.seek(0)` and writing still appends to the end of the file. So I think it's probably fine to just append to the end of the file. Or maybe it's not even necessary to append, just truncate.